src/eval/emit_js.rs

```diff
@@ -0,0 +1,419 @@
+        ast::Expr::Select(select) => emit_select_expr(select, out_src),
+        ast::Expr::Str(string) => emit_string_expr(string, out_src),
+        ast::Expr::UnaryOp(unary_op) => emit_unary_op(unary_op, out_src),
+        ast::Expr::With(with) => emit_with(with, out_src),
+        _ => panic!("emit_expr: not implemented: {:?}", nix_ast),
+    }
+}
+
+fn emit_apply(apply: &ast::Apply, out_src: &mut String) -> Result<(), String> {
+    emit_expr(
+        &apply
+            .lambda()
+            .expect("Unexpected lambda application without the lambda."),
+        out_src,
+    )?;
+    out_src.push_str(".apply(");
+    emit_expr(
+        &apply
+            .argument()
+            .expect("Unexpected lambda application without arguments."),
+        out_src,
+    )?;
+    out_src.push(')');
+    Ok(())
+}
+
+fn emit_attrset(attrset: &ast::AttrSet, out_src: &mut String) -> Result<(), String> {
+    emit_has_entry(attrset, attrset.rec_token().is_some(), out_src)
+}
+
+fn emit_has_entry(
+    has_entry: &impl ast::HasEntry,
+    is_recursive: bool,
+    out_src: &mut String,
+) -> Result<(), String> {
+    *out_src += "n.";
+    *out_src += if is_recursive {
+        "recAttrset"
+    } else {
+        "attrset"
+    };
+    *out_src += "(ctx,(ctx) => [";
+    for attrpath_value in has_entry.attrpath_values() {
+        out_src.push('[');
+        let attrpath = attrpath_value.attrpath().expect("Not implemented");
+        let value = &attrpath_value.value().expect("Not implemented");
+        emit_attrpath(&attrpath, out_src)?;
+        *out_src += ",new n.Lazy(ctx,(ctx) => ";
+        emit_expr(value, out_src)?;
+        *out_src += ")],";
+    }
+    *out_src += "])";
+    Ok(())
+}
+
+fn emit_attrpath(attrpath: &ast::Attrpath, out_src: &mut String) -> Result<(), String> {
+    *out_src += "[";
+    for attr in attrpath.attrs() {
+        out_src.push_str("new n.Lazy(ctx,(ctx) =>");
+        match attr {
+            ast::Attr::Ident(ident) => {
+                emit_nix_string(ident.ident_token().expect("Missing token.").text(), out_src)
+            }
+            ast::Attr::Str(str_expression) => emit_string_expr(&str_expression, out_src)?,
+            ast::Attr::Dynamic(expr) => {
+                emit_expr(&expr.expr().expect("Expected an expression."), out_src)?
+            }
+        }
+        out_src.push_str("),");
+    }
+    *out_src += "]";
+    Ok(())
+}
+
+fn emit_nix_string(string: &str, out_src: &mut String) {
+    *out_src += "new n.NixString(\"";
+    *out_src += string;
+    *out_src += "\")";
+}
+
+fn emit_bin_op(bin_op: &ast::BinOp, out_src: &mut String) -> Result<(), String> {
+    let operator = bin_op.operator().expect("Not implemented");
+    let lhs = &bin_op.lhs().expect("Not implemented");
+    let rhs = &bin_op.rhs().expect("Not implemented");
+    match operator {
+        // Arithmetic
+        // ast::BinOpKind::Add => emit_add_bin_op(lhs, rhs, out_src)?,
```

Comment:
Oh, this is my bad, I forgot to remove this comment. Would you mind removing it while you're at it?
```suggestion
```